### DISCUSSION TITLE
Delete a tile from another layer with automap

### DISCUSSION THREAD

**Discussion by duarm** [mapeditor/tiled#4024] [Q&A] [answered]
Assume the player is in layer 1, I have the following scenario

![image](https://github.com/user-attachments/assets/008a2f65-777f-42d2-9c9c-5f46b920146d)

I want the player to pass through a door, so the layer 0 needs to be behind, and layer 2 above, as expected. Translating that to automap, I have this

![image](https://github.com/user-attachments/assets/e0e6fb21-c516-41f2-acb6-6aed3253feef)
![image](https://github.com/user-attachments/assets/9f25fffb-7ff9-4581-8697-2d43c71f87d7)

The first rule matches loosely a lot of situations, but then the second rule, which is applied after the first one, overrides the first one where necessary, resulting in the desired outcome. Now, if I output the first rule to another layer, on our case, layer 2, the second does not override the first one anymore, resulting in something like this

![image](https://github.com/user-attachments/assets/b73cae1a-b199-4655-829f-855215cd196d)

which is expected. I'm solving this by outputting an special tile in the second layer, which my engine recognizes and ignores, like this  (the "delete" special tile signals the engine to ignore that tile)
![image](https://github.com/user-attachments/assets/f89a5a34-1bc2-4a4f-a6ec-edec4078383d)

Is there a tiled solution? It would be nice if we had a way to tell tiled to delete a tile from another layer instead of overriding it with a dummy tile.

**Comment by bjorn** [comment, marked as answer]
It sounds like you'll want to try outputting the "Empty" [special tile](https://doc.mapeditor.org/de/latest/manual/automapping/#specialtiles). The documentation states:

> If used on an output layer, this tile will output an empty tile, allowing you to erase tiles with Automapping.

**Reply by duarm** [reply]
I missed that, Thanks. Tried the negate tile, but the Empty tile ended up being the one I wanted.
